### PR TITLE
Annotations of intertrial choices, cue_switch, and iti_switch

### LINKED ISSUE #24: annotations should add label for "within session"
The mice often lick for several minutes while the session is getting set up. We should add a reliable label that excludes and licks/bouts that happen before the first go cue, and after the last go cue.

### PR DESCRIPTION
**Pull request by alexpiet**
- [x] Adds annotations of intertrial choices, cue switch and iti switch
- [x] All annotations can be included with `df_licks = annotate_licks(nwb)`
- [x] plot_interlick_interval() is a function for easily plotting interlick intervals
- [x] update session scroller
   - [x] Need to add a check for reward annotations, since right now we get an error
   - [x] Adopt a general approach of "if the annotation is provided, we plot it"
   - [x] intertrial_choice
- [x] Add significant unit test overall. Achieve 100% coverage with separate tests for important cases
- [x] Add annotations for "within session" (resolves #24)

Screenshot 2024-10-11 at 11 37 02 AM

### CODE REVIEW

**Comment by alexpiet** [comment]
@rachelstephlee @hanhou @hagikent @ZhixiaoSu This is ready for review

File: src/aind_dynamic_foraging_basic_analysis/licks/annotation.py [hunk 1]
```diff
@@ -157,3 +185,141 @@ def annotate_cue_response(nwb):
...
+    temp = temp.reset_index().set_index("index")
+    df_licks["bout_intertrial_choice"] = temp["bout_intertrial_choice"]
+
+    return df_licks
+
+
+def annotate_switches(nwb):
+    """
+    cue_switch: this cue_choice differs from the previous cue_choice
+    iti_switch: this intertrial_choice differs from the previous choice (iti or cue)
```
[the remark this answers is not in the document]
Comment (alexpiet) [reply src/aind_dynamic_foraging_basic_analysis/licks/annotation.py:229]: No. the iti_switch is defined as whether this intertrial_choice differs from the previous choice, regardless of whether it was an iti_choice or cue_choice

[the remark this answers is not in the document]
Comment (rachelstephlee) [reply src/aind_dynamic_foraging_basic_analysis/licks/annotation.py:229]: got it thanks !

**Review by rachelstephlee — COMMENTED** [review]
looks good. thanks for the addition.

**Comment by alexpiet** [comment]
> looks good. thanks for the addition.

Can you approve the PR?

**Review by rachelstephlee — APPROVED** [review]

**Pull request merged**